src/transformers/tokenization_utils_base.py

```diff
@@ -1150,35 +1150,35 @@ def additional_special_tokens_ids(self) -> List[int]:
 
     @bos_token_id.setter
     def bos_token_id(self, value):
-        self._bos_token = self.convert_tokens_to_ids(value)
+        self._bos_token = self.convert_ids_to_tokens(value) if value is not None else None
```

Comment:
Nice! But wouldn't it be better to let it fail when the `value` is `None` ?  Because if one sets it to a `None` value and it doesn't fail, one would assume the token is set and try to use and then it'll fail in some later part of the code. It's better to fail as early as possible IMO. wdyt @SaulLu ?

Reply:
From my side, I think that some models don't need to define some/all special tokens (bos, eos, etc). It seems fine to me that the id can be set to None. :slightly_smiling_face: 

Besides, we can see in the property `bos_token_id` that a `None` value is well handled:
https://github.com/huggingface/transformers/blob/12bfa97a4373b7b883ee0de4f2c9c53918d81cf1/src/transformers/tokenization_utils_base.py#L1066-L1074